exonum/src/api_ng/node/private/mod.rs

```diff
@@ -0,0 +1,213 @@
+    Active,
+    Reconnect(ReconnectInfo),
+}
+
+impl Default for IncomingConnectionState {
+    fn default() -> IncomingConnectionState {
+        IncomingConnectionState::Active
+    }
+}
+
+#[derive(Serialize, Deserialize, Default)]
+struct IncomingConnection {
+    public_key: Option<PublicKey>,
+    state: IncomingConnectionState,
+}
+
+#[derive(Serialize, Deserialize)]
+struct PeersInfo {
+    incoming_connections: Vec<SocketAddr>,
+    outgoing_connections: HashMap<SocketAddr, IncomingConnection>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+struct PeerAddQuery {
+    ip: SocketAddr,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+struct ConsensusEnabledQuery {
+    enabled: bool,
+}
+
+/// Private system API.
+#[derive(Clone, Debug)]
+pub struct SystemApi {
+    info: NodeInfo,
+    shared_api_state: SharedNodeState,
+}
+
+impl SystemApi {
+    /// Creates a new `private::SystemApi` instance.
+    pub fn new(info: NodeInfo, shared_api_state: SharedNodeState) -> SystemApi {
+        SystemApi {
+            info,
+            shared_api_state,
+        }
+    }
+
+    /// Adds private system API endpoints to the corresponding scope.
+    pub fn wire(self, api_scope: &mut ServiceApiScope) -> &mut ServiceApiScope {
+        self.handle_peers_info("v1/peers", api_scope)
+            .handle_peer_add("v1/peers", api_scope)
+            .handle_network_info("v1/network", api_scope)
+            .handle_is_consensus_enabled("v1/consensus_enabled", api_scope)
+            .handle_set_consensus_enabled("v1/consensus_enabled", api_scope)
+            .handle_shutdown("v1/shutdown", api_scope);
+        api_scope
+    }
+
+    fn handle_peers_info(self, name: &'static str, api_scope: &mut ServiceApiScope) -> Self {
+        let self_ = self.clone();
+        api_scope.endpoint(name, move |_state: &ServiceApiState, _query: ()| {
+            let mut outgoing_connections: HashMap<SocketAddr, IncomingConnection> = HashMap::new();
+
+            for socket in self.shared_api_state.outgoing_connections() {
+                outgoing_connections.insert(socket, Default::default());
+            }
+
+            for (s, delay) in self.shared_api_state.reconnects_timeout() {
+                outgoing_connections
+                    .entry(s)
+                    .or_insert_with(Default::default)
+                    .state = IncomingConnectionState::Reconnect(ReconnectInfo { delay });
+            }
+
+            for (s, p) in self.shared_api_state.peers_info() {
+                outgoing_connections
+                    .entry(s)
+                    .or_insert_with(Default::default)
+                    .public_key = Some(p);
+            }
+
+            Ok(PeersInfo {
+                incoming_connections: self.shared_api_state.incoming_connections(),
+                outgoing_connections,
+            })
+        });
+        self_
+    }
+
+    fn handle_peer_add(self, name: &'static str, api_scope: &mut ServiceApiScope) -> Self {
```

Comment:
Why not using constants instead of `name` arguments for these functions?

Reply:
With `name` [this](https://github.com/alekseysidorov/exonum/blob/actix-web-api/exonum/src/api_ng/node/private/mod.rs#L119) code looks more clear.